code/game/objects/items/pinpointer.dm

```diff
@@ -1,143 +1,114 @@
 /obj/item/pinpointer
-	name = "pinpointer"
-	icon_state = "pinoff"
-	flags_atom = CONDUCT
-	flags_equip_slot = ITEM_SLOT_BELT
-	w_class = 1
-	item_state = "electronic"
-	throw_speed = 4
-	throw_range = 20
-	matter = list("metal" = 500)
-	var/obj/item/disk/nuclear/the_disk = null
-	var/active = 0
-
-
-	attack_self()
-		if(!active)
-			active = 1
-			workdisk()
-			to_chat(usr, "<span class='notice'>You activate the pinpointer</span>")
-		else
-			active = 0
-			icon_state = "pinoff"
-			to_chat(usr, "<span class='notice'>You deactivate the pinpointer</span>")
-
-	proc/workdisk()
-		if(!active) return
-		if(!the_disk)
-			the_disk = locate()
-			if(!the_disk)
-				icon_state = "pinonnull"
-				return
-		setDir(get_dir(src,the_disk))
-		switch(get_dist(src,the_disk))
-			if(0)
-				icon_state = "pinondirect"
-			if(1 to 8)
-				icon_state = "pinonclose"
-			if(9 to 16)
-				icon_state = "pinonmedium"
-			if(16 to INFINITY)
-				icon_state = "pinonfar"
-		spawn(5) .()
-
-	examine(mob/user)
-		..()
-		for(var/obj/machinery/nuclearbomb/bomb in GLOB.machines)
-			if(bomb.timing)
-				to_chat(user, "Extreme danger.  Arming signal detected.   Time remaining: [bomb.timeleft]")
+    name = "pinpointer"
+    icon_state = "pinoff"
+    flags_atom = CONDUCT
+    flags_equip_slot = ITEM_SLOT_BELT
+    w_class = 1
+    item_state = "electronic"
+    throw_speed = 4
+    throw_range = 20
+    matter = list("metal" = 500)
+    var/active = FALSE
+    var/target = null
+
+/obj/item/pinpointer/proc/set_target()
+    if (issurvivorgamemode(SSticker.mode))
```

Comment:
Early return

Reply:
Thats not possible for this proc, unless you can explain it a little more?